protocol/tx/spend.go

```diff
@@ -9,17 +9,43 @@ type spend struct {
 		ExtHash     bc.Hash
 	}
 	ordinal int
+
+	// SpentOutput contains (a pointer to) the manifested entry
+	// corresponding to body.SpentOutput. It may be nil, in which case
+	// spentInfo must be non-nil.
+	SpentOutput *output
+
+	// spentInfo contains validation-essential data in case the spent
+	// output object is not present (in SpentOutput).
```

Comment:
> We can't construct a full-fledged `tx.output` from a `bc.TxInput`

Right ok, that's what I was thinking of when I said "artifact of the mapping logic". Once we've converted everything to txgraph, there will be no bc.TxInput, and the transaction will contain a full copy of the previous tx's output entry right? In that case, I would think we won't need this spentInfo type any more.